src/codegen/generate.ts

```diff
@@ -360,11 +364,48 @@ export default class ApiGenerator {
 
   /**
    * Create a type alias for the schema referenced by the given ReferenceObject
+   * @param ignoreDiscriminator If true, the discriminator property of the schema referenced by
+   * `obj` will be ignored. This is meant to be used when getting the type of a parent schema in an
+   * `allOf` constructor.
    */
-  getRefAlias(obj: OpenAPIV3.ReferenceObject, onlyMode?: OnlyMode) {
-    const { $ref } = obj;
+  getRefAlias(
+    obj: OpenAPIV3.ReferenceObject,
+    onlyMode?: OnlyMode,
+    ignoreDiscriminator?: boolean,
+  ) {
+    const available = ($ref: string) => {
+      try {
+        this.resolve({ $ref });
+        return false;
+      } catch (error) {
+        return true;
+      }
+    };
+
+    const $ref = ((
+      obj: OpenAPIV3.ReferenceObject,
+      ignoreDiscriminator: boolean,
+    ) => {
+      let { $ref } = obj;
+      if (!ignoreDiscriminator) return $ref;
+
+      $ref += "Base";
+      if (available($ref)) return $ref;
+
+      let i = 2;
+      while (true) {
+        const $ref_ = $ref + String(i);
+        if (available($ref_)) return $ref_;
+        i += 1;
+      }
+    })(obj, ignoreDiscriminator ?? false);
```

Comment:
Can we move this into a standalone function? like `findAvailableRefName`?

Reply:
OK. Maybe we should name the return value of `findAvailableRefName` something other than `$ref`, since it's no longer the `$ref` property of the `obj`. How about simply using `key`?

Reply:
yeah, sounds feasible to me 👍

Reply:
I've decided not to change the name after all.